src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -292,20 +337,79 @@ void java_bytecode_convert_classt::convert(const classt &c)
     throw 0;
   }
 
-  // now do fields
+  // Now do fields
+  for(auto overlay_class : overlay_classes)
+  {
+    for(const auto &field : overlay_class.get().fields)
+    {
+      debug()
+        << "Adding symbol from overlay class:  field '" << field.name << "'"
+        << eom;
+      convert(*class_symbol, field);
+    }
+  }
   for(const auto &field : c.fields)
   {
     debug() << "Adding symbol:  field '" << field.name << "'" << eom;
     convert(*class_symbol, field);
   }
 
-  // now do methods
-  for(const auto &method : c.methods)
+  // Now do methods
+  std::set<irep_idt> overlay_methods;
+  for(auto overlay_class : overlay_classes)
+  {
+    for(const methodt &method : overlay_class.get().methods)
+    {
+      const irep_idt method_identifier =
+        id2string(qualified_classname) + "." + id2string(method.name)
+          + ":" + method.descriptor;
+      if(method_bytecode.contains_method(method_identifier))
+      {
+        if(overlay_methods.count(method_identifier) == 0)
+        {
+          warning()
```

Comment:
I don't understand this warning. The conditions above as I understand it say:

 - we're looking at a method from an overlay class 
 - the method also exists in the original class
 - the method has not already been seen on any overlay for this specific class

But your warning seems to suggest:
 - is_overlay_method returned false 
 - and it _has_ been seen already i.e. `overlay_methods.count(method_identifier) > 0` 

Not sure which is correct but I think something isn't matching up here.